controllers/datadogagent/feature/admissioncontroller/feature_test.go

```diff
@@ -52,49 +38,152 @@ func TestAdmissionControllerFeature(t *testing.T) {
 			WantConfigure: true,
 			ClusterAgent:  testDCAResources("hostip", "", false),
 		},
-		//////////////////////////
-		// v2Alpha1.DatadogAgent
-		//////////////////////////
+		// //////////////////////////
+		// // v2alpha1.DatadogAgent
+		// //////////////////////////
 		{
-			Name:          "v2alpha1 admission controller not enabled",
-			DDAv2:         newV2Agent(false, "", "", false, &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}, nil),
+			Name: "v2alpha1 Admission Controller not enabled",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				Build(),
 			WantConfigure: false,
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled",
-			DDAv2:         newV2Agent(true, "", "", false, &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}, nil),
+			Name: "v2alpha1 Admission Controller enabled with basic setup",
+			DDAv2: v2alpha1test.NewDatadogAgentBuilder().
+				WithAdmissionControllerEnabled(true).
+				Build(),
 			WantConfigure: true,
-			ClusterAgent:  testDCAResources("", "", false),
+			ClusterAgent: test.NewDefaultComponentTest().WithWantFunc(
+				admissionControllerWantFunc("", "")),
 		},
 		{
-			Name:          "v2alpha1 admission controller enabled, apm uses uds",
-			DDAv2:         newV2Agent(true, "", "", false, apmUDS, &v2alpha1.DogstatsdFeatureConfig{}, nil),
+			Name:          "v2alpha1 admission controller enabled, cwsInstrumentation enabled",
+			DDAv2:         newV2Agent(true, "", "", true, &v2alpha1.APMFeatureConfig{}, &v2alpha1.DogstatsdFeatureConfig{}, nil),
```

Comment:
nit - this uses old approach and brings back two functions used only by this test. Can be fixed in another PR since this was added in main.